src/treeinterp/BTFactory.cpp

```diff
@@ -19,6 +28,21 @@ void BTFactory::makeTrees() {
 
     std::cout << "Re-Make Trees From Json" << std::endl;
 
+    // We need this instance of the TreeProtoType Class to use its methods. An alternative would be to make the methods static,
+    // but this essentially makes them global which is dangerous and unnecessary in my opinion. Additionally, some settings
+    // are initialized in the default constructor of TreeProtoType
```

Comment:
Comments indeed should explain why you made certain decisions :+1: rather than how things work.